ecosystem/typescript/sdk_v2/src/internal/general.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/general}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * general namespace and without having a dependency cycle error.
+ */
+
+import { AptosConfig } from "../api";
+import { get, post } from "../client";
+import { Block, LedgerInfo, LedgerVersion, MoveValue, TableItemRequest, ViewRequest } from "../types";
+import { AptosApiType } from "../utils/const";
+
+export async function getLedgerInfo(args: { aptosConfig: AptosConfig }): Promise<LedgerInfo> {
+  const { aptosConfig } = args;
+  const { data } = await get<{}, LedgerInfo>({
+    url: aptosConfig.getRequestUrl(AptosApiType.FULLNODE),
+    endpoint: "",
+    originMethod: "getLedgerInfo",
+    overrides: { ...aptosConfig.clientConfig },
+  });
+  return data;
+}
+
+export async function getBlockByVersion(args: {
+  aptosConfig: AptosConfig;
+  blockVersion: number;
+  options?: { withTransactions?: boolean };
+}): Promise<Block> {
+  const { aptosConfig, blockVersion, options } = args;
+  const { data } = await get<{}, Block>({
+    url: aptosConfig.getRequestUrl(AptosApiType.FULLNODE),
+    endpoint: `blocks/by_version/${blockVersion}`,
+    originMethod: "getBlockByVersion",
+    params: { with_transactions: options?.withTransactions },
+    overrides: { ...aptosConfig.clientConfig },
+  });
+  return data;
```

Comment:
I see a great place for consolidation, where all fullnode gets could be reduced to:

```
getFullnode(
  {
    aptosConfig: AptosConfig,
    endpoint: string,
    name: string,
    params: {},
    overrides: {}
  }
)
```